### PR TITLE
GFilePicker: Added optional file extension param

### PR DESCRIPTION
**Pull request by rhin123**
This seems so simple but I have a slight feeling I did something wrong here... no?

### CODE REVIEW

File: Libraries/LibGUI/GFilePicker.cpp [hunk 1]
```diff
@@ -128,7 +129,7 @@ GFilePicker::GFilePicker(Mode mode, const StringView& path, CObject* parent)
     filename_label->set_preferred_size(60, 0);
     auto* filename_textbox = new GTextBox(filename_container);
     if (m_mode == Mode::Save) {
-        filename_textbox->set_text("Untitled.txt"); //TODO: replace .txt with a preferred extension
+        filename_textbox->set_text(String::format("%s.%s", "Unitled", m_file_extension.characters()));
```
Comment (awesomekling) [inline Libraries/LibGUI/GFilePicker.cpp]: s/Unitled/Untitled/

File: Libraries/LibGUI/GFilePicker.h [hunk 2]
```diff
@@ -15,10 +15,10 @@ class GFilePicker final : public GDialog {
     };
 
     static Optional<String> get_open_filepath();
-    static Optional<String> get_save_filepath();
+    static Optional<String> get_save_filepath(const String& m_file_extension);
```
Comment (awesomekling) [inline Libraries/LibGUI/GFilePicker.h]: The `m_` prefix is for member variables, so please don't use it on an argument name.

**Comment by rhin123** [comment]
> Looks fine, what did you think would be harder?

I guess it's the bad habit for over-complicating things. Oh well

**Comment by awesomekling** [comment]
Oh, this has a conflict with your other change. Could you fix it?
And next time feel free to do multiple commits like this in a single PR, since these are all indeed QoL improvements for GFilePicker. :)

**Comment by rhin123** [comment]
JESUS i hate git

**Pull request closed without merging**